FrameworkSystem/Client/WebAppCompiler.py

```diff
@@ -0,0 +1,353 @@
+"""
+It is used to compile the web framework
+"""
+
+
+__RCSID__ = "$Id$"
+
+import os, tempfile, shutil, subprocess, gzip, sys, imp
+
+from DIRAC import gLogger, gConfig, rootPath, S_OK, S_ERROR
+from DIRAC.Core.Utilities.CFG import CFG
+
+class WebAppCompiler( object ):
+  
+  def __init__( self, params ):
+    
+    self.__params = params
+    
+    self.__extVersion = '4.2.1.883'
+    self.__extDir = 'extjs'  # this directory will contain all the resources required by ExtJS
+    
+    self.__sdkDir = params.extjspath if self.__params.extjspath is not None else '/opt/dirac/extjs/ext-4.2.1.883'
+    
+     
+    self.__webAppPath = os.path.join( self.__params.destination, 'WebAppDIRAC', 'WebApp' )
+    self.__staticPaths = [os.path.join( self.__webAppPath, 'static' )]
+    if self.__params.name != 'WebAppDIRAC':
+      self.__staticPaths.append( os.path.join( self.__params.destination, self.__params.name, 'WebApp', 'static' ) )
+                           
+    self.__classPaths = [ os.path.join( self.__webAppPath, *p ) for p in ( ( "static", "core", "js", "utils" ),
+                                                                           ( "static", "core", "js", "core" ) )]
+    
+    
+    self.__classPaths.append( os.path.join( os.path.dirname( self.__sdkDir ), "examples", "ux" ) )
+    self.__classPaths.append( os.path.join( os.path.dirname( self.__sdkDir ), "examples", "ux", "form" ) )
+    
+    self.__sdkPath = os.path.join( self.__sdkDir, "src" )
+    
+    self.__extjsDirsToCopy = [os.path.join( os.path.dirname( self.__sdkDir ), "resources" )]    
+    self.__extjsFilesToCopy = [os.path.join( os.path.dirname( self.__sdkDir ), "ext-all-dev.js" )]   
+    
+    self.__debugFlag = str( gLogger.getLevel() in ( 'DEBUG', 'VERBOSE', 'INFO' ) ).lower()
+    self.__compileTemplate = os.path.join( self.__params.destination, 'WebAppDIRAC', "Lib", "CompileTemplates" )
+    
+    self.__senchacmddir = os.path.join( rootPath, "sbin", "Sencha", "Cmd" )  # this place will be used, if sencha cmd is not available
+    self.__senchaVersion = "v6.5.0.180"
```

Comment:
what about a parameter?

Reply:
This is again for backward compatibility. This can be ignored as well.